app/src/main/java/com/github/se/icebreakrr/ui/tags/TagView.kt

```diff
@@ -35,179 +34,132 @@ import androidx.compose.ui.unit.TextUnit
   }
 }
 
-/**
- * This Composable is a row of tags that are clickable and that adapts dynamically to the space it
- * has available
- *
- * @param l: The list of tags to show
- * @param tagStyle: the style of your tags
- * @param onClick: the event when the user clicks on a tag
- */
 @OptIn(ExperimentalLayoutApi::class)
 @Composable
-fun RowOfClickTags(l: List<Pair<String, Color>>, tagStyle: TagStyle, onClick: (String) -> Unit) {
+fun RowOfClickTags(tags: List<Pair<String, Color>>, tagStyle: TagStyle, onClick: (String) -> Unit) {
   val isExtended = remember { mutableStateOf(false) }
   LazyColumn(modifier = Modifier.fillMaxSize()) {
     item {
       FlowRow(
-          modifier = Modifier.padding(8.dp),
+          modifier = Modifier.padding(TAG_PADDING),
           horizontalArrangement = Arrangement.Start,
           verticalArrangement = Arrangement.Top) {
-            if (isExtended.value) {
-              l.forEach { pair ->
-                ClickTag(pair.first, TagStyle(tagStyle.textColor, pair.second, tagStyle.fontSize)) {
-                  onClick(pair.first)
-                }
-              }
-            } else {
-              val notExtendedList = l.take(3)
-              notExtendedList.forEach { pair ->
-                ClickTag(pair.first, TagStyle(tagStyle.textColor, pair.second, tagStyle.fontSize)) {
-                  onClick(pair.first)
-                }
-              }
-              if (notExtendedList.size >= 3) {
-                ExtendTag(tagStyle) { isExtended.value = true }
+            val tagsToShow = if (isExtended.value) tags else tags.take(3)
+            tagsToShow.forEach { (text, color) ->
+              ClickTag(text, TagStyle(tagStyle.textColor, color, tagStyle.fontSize)) {
+                onClick(text)
               }
             }
+            if (!isExtended.value && tags.size > 3) {
+              ExtendTag(tagStyle) { isExtended.value = true }
+            }
           }
     }
   }
 }
 
-/**
- * This Composable is used in the Edit Profile, allows a user to enter a text in a text field ant to
- * get a list of tags to choose from We also get a collection of all the tags we have already
- * selected
- *
- * @param selectedTag : all the tags in the that have already been selected
- * @param outputTag : the tags that drop down in the DropDownMenu
- * @param stringQuery : The text that is modified by the user
- * @param expanded : Choose if the drop down menu is activated or not
- * @param onTagClick : The event happening when the user clicks on a tag
- * @param onDropDownItemClicked : The event when the user clicks on a DropDownMenuItem
- * @param onStringChanged : The event when the string is changed by the user
- * @param textColor : The color of the text we want in the tags selected and on the tags in the drop
- *   down menu
- * @param textSize : The size of the text in the tags selected and on the tags in the drop down menu
- *   The color in the tag depends on the category of the tag
- */
```

Comment:
why did you get rid of the documentation?

Reply:
yes this was an error again